### ISSUE TITLE
Snippet: A Request_and_Probable Bug(?)_and_Elaboration

### ISSUE THREAD

**Issue by Ninimama** [matrix-profile-foundation/matrixprofile#23]
Hello,

Thanks for creating this amazing package. It helps others to use it for different applications. 
I have three things to discuss regarding the snippets:

(1) It would be great if the locations of **all** snippets could be returned as well. (By that, I mean all the subsequences that are close to a snippet which is considered as the representative of those subsequences). 

(2) "Probable" Bug: I have an hourly data recorded in a year (array: 8760-hr time series). I tried to find two snippets (with length 24, and window size=12) and I got a result. Then, I just deleted the first 12 samples (i.e. new_array = array[12:]). Rationally speaking, there should be no change in the snippets, but that is not the case as the results have been changed drastically. I also tried array[1:] and the result was the same as the first one but with one hour shift. 
I noticed that, in all three cases, the index (given in the result), from which the snippet starts, is multiple of 24 (my snippet size) which might be a little bit strange. 

(3) I tried to read the snippet paper published by Dr. Eamon Keogh's team. I found the subsequent length m, the snippet size. However, I couldn't find the purpose of window size, the one used in the snippet function in this package. It would be great if you could give some elaboration on this matter.

Best,
Nima

**Comment by tylerwmarrs** [comment]
I have not lost sight of this issue and plan to create a few open issues that I would love to have you contribute to.

1. Add support for neighboring snippet indices
2. Add visualization for snippets

Would you be interested in contributing? 

In regards to your issue, please upload any relevant data (if possible) or data visualizations. This will help paint the picture.

The window size is the same as m in the paper. We tend to be more explicit with parameter and variable names for readability purposes.

**Comment by tylerwmarrs** [comment]
@Ninimama I created the open issues. Are you able to provide code for the feature requests?

**Comment by Ninimama** [comment]
Hi @tylerwmarrs,

Sorry for my late response. 

I would love to contribute; however, I am a newbie in GitHub and programming. I think there is a difference between "coding" and "efficient & stylish coding." 

Although I mostly look for the applications of different methods in my field (electrical engineering), I am really eager to learn more about using GitHub, [efficient] coding, and contributing to projects.

Is that okay if I see you as a mentor for now and ask for a guideline for getting into Github and Contributing to different projects? (If yes, where can I send you a short email regarding the questions I have?)

I will upload a few figures about the snippets soon.

Best,
Nima

**Comment by tylerwmarrs** [comment]
The best way to reach me directly is through our Discord channel. https://discordapp.com/invite/sBhDNXT

You can submit a partial pull request for any fix and I can help tidy up the code and documentation is necessary.

**Comment by vanbenschoten** [comment]
@Ninimama I wanted to reach back out regarding this issue. Please let us know how we can assist you in making a pull request! We greatly appreciate your help :)

**Comment by Ninimama** [comment]
@vanbenschoten 

Very sorry for my very late response. I've been working on my research. I am still struggling for publishing my first paper :) 

I provided some of the results I obtained when I was using a matrix profile in issue #25 

I hope I can finish my paper soon and get back here to work on this. 

Is there any good tutorial in helping me understand the basics of Github and how it works? like the pull request and ...? I haven't used Github that much and I am in the Electrical Engineering field, but I would like to understand some of the basics. (Since I am actually trying to switch my field to ML and I am practicing programming almost daily as well) So, I can read it while I am finishing my paper, and then I can get back here. 

Any suggestion?
